### PR TITLE
Remove Nebari-HPC docs

### PR DESCRIPTION
**Pull request by pavithraes**
Nebari-HPC docs were moved during the QHub->Nebari migration, but we don't actually have a Nebari-HPC project to share right now.

This PR removed the Nebari-HPC docs and all mentions of HPC support.

We can add these doc back and update them, when there is a stable user workflow for HPC deployments :)

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,8 +28,8 @@ To get started with Nebari right away, check the [Nebari documentation][nebari-d
 
 ## Table of contents
 
+- [Table of contents](#table-of-contents)
```
Comment (pavithraes) [inline README.md:31]:
```suggestion
```

**Comment by pavithraes** [comment]
@viniciusdc Would you mind taking a look at this when you get a chance? Thanks!

**Review by viniciusdc — APPROVED** [review]
LGTM, thanks @pavithraes

**Comment by viniciusdc** [comment]
Once this is merged we will follow up the work on https://github.com/nebari-dev/nebari-docs/issues/435

**Pull request merged**